packages/lib/passport-interface/src/RequestTypes.ts

```diff
@@ -1211,3 +1211,74 @@ export type GenericIssuanceSendPipelineEmailRequest = {
 export type GenericIssuanceSendPipelineEmailResponseValue = {
   queued: number;
 };
+
+export enum EmailUpdateError {
+  InvalidCredential = "InvalidCredential",
+  InvalidConfirmationCode = "InvalidConfirmationCode",
+  EmailAlreadyRegistered = "EmailAlreadyRegistered",
+  CantDeleteOnlyEmail = "CantDeleteOnlyEmail",
+  CantChangeWhenMultipleEmails = "CantChangeWhenMultipleEmails",
+  EmailAlreadyAdded = "EmailAlreadyAdded",
+  EmailNotAssociatedWithThisAccount = "EmailNotAssociatedWithThisAccount",
+  UserNotFound = "UserNotFound",
+  InvalidInput = "InvalidInput",
+  TooManyEmails = "TooManyEmails",
+  Unknown = "Unknown"
+}
+
+export interface AddUserEmailRequest {
+  newEmail: string;
+
+  /**
+   * A semaphore signature from the user, used to verify their identity.
+   */
+  pcd: SerializedPCD<SemaphoreSignaturePCD>;
+
+  /**
+   * If absent, requests a confirmation code; if present, redeems it.
+   */
+  confirmationCode?: string;
+}
+
+export type AddUserEmailResponseValue =
+  | {
+      sentToken: false;
+      token?: never;
+      newEmailList: string[];
+    }
+  | { sentToken: true; token?: string; newEmailList?: never };
+
+export interface ChangeUserEmailRequest {
+  newEmail: string;
```

Comment:
This might be a good place to send the oldEmail to avoid confusion in race conditions.